File: datastructures/strings/python/algorithms/reverse_string.py
def reverseString(s):
    i = 0
    j = len(s) - 1

    while i < j:
        temp = s[i]
        s[i] = s[j]
        s[j] = temp
        i += 1
        j -= 1


def main():
    s = ['h', 'e', 'l', 'l', 'o']
    reverseString(s)
    print(s)


if __name__ == "__main__":
    main()

File: datastructures/strings/java/algorithms/ReverseString.java
import java.util.Arrays;

public class ReverseString {

    public static void reverseString(char[] s) {
        int i = 0;
        int j = s.length - 1;

        while (i < j) {
            char temp = s[i];
            s[i] = s[j];
            s[j] = temp;
            i++;
            j--;
        }
    }

    public static void main(String[] args) {
        char[] s = { 'h', 'e', 'l', 'l', 'o' };

        reverseString(s);

        Arrays.asList(s).stream().forEach(System.out::print);
    }

}
